reactor-core/src/main/java/reactor/util/Metrics.java

```diff
@@ -48,30 +59,89 @@ public class Metrics {
 	 *
 	 * @return true if the Micrometer instrumentation facade is available
 	 */
+	@Deprecated
 	public static final boolean isInstrumentationAvailable() {
 		return isMicrometerAvailable;
 	}
 
+	@Deprecated
 	public static class MicrometerConfiguration {
 
-		private static MeterRegistry registry = globalRegistry;
+		private static final Logger LOGGER = Loggers.getLogger(MicrometerHttpClientInterceptor.class);
```

Comment:
I was trying to dynamically discover the extra module and delegate to it without a dependency cycle, but that's a bit too much. will probably revert to duplicating code, with the subtlety that if the module is detected then `Metrics.isInstrumentationAvailable()` would now return false (ie. only true if and only if micrometer alone is on the classpath)